### PR TITLE
meta: Fix ah and bz search not working without cookie

### PR DESCRIPTION
**Pull request by NopoTheGamer**

### CODE REVIEW

File: src/main/java/io/github/moulberry/notenoughupdates/overlays/AuctionSearchOverlay.java [hunk 1]
```diff
@@ -333,8 +362,54 @@ public static void close() {
...
+							line1.append(word);
+							if (line1.length() >= 15) {
+								break;
+							} else {
+								line1.append(" ");
+							}
+						}
+					}
+					if (line1.length() >= 15) break;
+				}
```
Comment (nea89o) [inline src/main/java/io/github/moulberry/notenoughupdates/overlays/AuctionSearchOverlay.java:403]:
```suggestion
				ListIterator<String> words = SPACE_SPLITTER.splitToList(search).listIterator();
				StringBuilder line0 = new StringBuilder();
				while (words.hasNext() ){
					String word = words.next();
					if (line0.length() + word.length() > 15) {
						words.previous();
						break;
					}
					line0.append(word).append(' ');
				}
				StringBuilder line1 = new StringBuilder();
				while (words.hasNext() ){
					String word = words.next();
					if (line1.length() + word.length() > 15) {
						break;
					}
					line1.append(word).append(' ');
				}
```

**Review by nea89o — CHANGES_REQUESTED** [review]
I really don't like that we mix the static and non static uses of this class. the rendering related methods (relating to the screen) should really be in its own class. Having a subscribe event on that same screen instance is just overloading that classes functionality.

I also really need to say that we cant just copy over code. OOP exists, we can just use `extends SearchOverlayScreen` and put most of the code in that one.

File: src/main/java/io/github/moulberry/notenoughupdates/overlays/AuctionSearchOverlay.java (hunk 1 above)
Comment (nea89o) [reply src/main/java/io/github/moulberry/notenoughupdates/overlays/AuctionSearchOverlay.java:403]: This whole thing could also be a single regex: `(.{0,15})( (.{0,15}))?` or something along those lines

**Comment by NopoTheGamer** [comment]
> we can just use `extends SearchOverlayScreen` and put most of the code in that one.

I sorta just moved all the GUI code into the 1 file, the individual overlays still have the subscribe events but I think it makes sense this way

**Review by nea89o — APPROVED** [review]

**Review by Lulonaut — APPROVED** [review]

**Pull request merged**